src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/RuntimeInformation.Windows.cs

```diff
@@ -42,7 +42,8 @@ public static unsafe Architecture OSArchitecture
                     // If we are running an x64 process on a non-x64 windows machine, we will report x64 as OS architecture.
                     //
                     // IsWow64Process2 is only available on Windows 10+, so we will perform run-time introspection via indirect load
-                    if (NativeLibrary.TryGetExport(NativeLibrary.Load(Interop.Libraries.Kernel32), "IsWow64Process2", out IntPtr isWow64Process2Ptr))
+                    IntPtr kernel32 = Interop.Kernel32.LoadLibraryEx(Interop.Libraries.Kernel32, 0, Interop.Kernel32.LOAD_LIBRARY_SEARCH_SYSTEM32);
```

Comment:
Unrelated change to make load library more secure?

Reply:
Yes: https://github.com/dotnet/runtime/pull/72452#discussion_r924619677